src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -14,43 +17,43 @@ namespace Microsoft.VisualStudio.LanguageServices.Experimentation
     [ExportWorkspaceService(typeof(IExperimentationService), ServiceLayer.Host), Shared]
     internal class VisualStudioExperimentationService : ForegroundThreadAffinitizedObject, IExperimentationService
     {
-        private readonly object _experimentationServiceOpt;
-        private readonly MethodInfo _isCachedFlightEnabledInfo;
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
+
+        private Optional<Func<string, bool>> _isCachedFlightEnabled;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public VisualStudioExperimentationService(IThreadingContext threadingContext, SVsServiceProvider serviceProvider)
-            : base(threadingContext, assertIsForeground: true)
+            : base(threadingContext)
         {
-            try
-            {
-                _experimentationServiceOpt = serviceProvider.GetService(typeof(SVsExperimentationService));
-                if (_experimentationServiceOpt != null)
-                {
-                    _isCachedFlightEnabledInfo = _experimentationServiceOpt.GetType().GetMethod(
-                        "IsCachedFlightEnabled", BindingFlags.Public | BindingFlags.Instance);
-                }
-            }
-            catch
-            {
-            }
+            _asyncServiceProvider = (IAsyncServiceProvider)serviceProvider;
         }
 
-        public bool IsExperimentEnabled(string experimentName)
+        public async ValueTask<bool> IsExperimentEnabledAsync(string experimentName, CancellationToken cancellationToken)
         {
-            ThisCanBeCalledOnAnyThread();
```

Comment:
How are we able to prevent additional calls on this that aren't using JTF? I agree this is a good and appropriate use in the Roslyn codebase, although I assume we've already asked the team that owns SVsExperimentationService what it would take to make it fully free-threaded? It's just that right now this is _special_ in our codebase and different than everything else, which almost guarantees a bug at some point.

Reply:
> How are we able to prevent additional calls on this that aren't using JTF?

I don't understand this question.

>  although I assume we've already asked the team that owns SVsExperimentationService what it would take to make it fully free-threaded

In this pull request, `SVsExperimentationService` is *used* under the assumption that it is already free-threaded. This behavior is carried over from the behavior before this pull request.

> It's just that right now this is special in our codebase and different than everything else, 

I don't understand this comment.